tests/openwisp2/sample_check/models.py

```diff
@@ -1,8 +1,21 @@
+from django.db import models
+from django.utils.timezone import now
 from openwisp_monitoring.check.base.models import AbstractCheck
 from swapper import swappable_setting
 
 
 class Check(AbstractCheck):
+    last_called = models.DateTimeField(blank=True, null=True)
+
     class Meta(AbstractCheck.Meta):
         abstract = False
         swappable = swappable_setting('check', 'Check')
+
+    def perform_check(self, store=True):
+        """
+        initiates check instance and calls its check method
+        """
+        self.last_called = now()
+        self.full_clean()
+        self.save()
+        return self.check_instance.check(store=True)
```

Comment:
I think this last line can be replaced by:

```python
return super().perform_check(store=store)
```

Which is more future proof.

Reply:
@nemesisdesign, can you please explain as to how exactly is it `Future Proof` so I can take note :smiley:

Reply:
Found you.

I meant calling `super()` instead of duplicating `self.check_instance.check(store=True)` , because so if the parent method is refactored or changed this method here won't need to be changed.